modules/plugins/assistant/codecompanion/config.nix

```diff
@@ -0,0 +1,26 @@
+{
+  config,
+  lib,
+  ...
+}: let
+  inherit (lib.modules) mkIf;
+
+  cfg = config.vim.assistant.codecompanion-nvim;
+in {
+  config = mkIf cfg.enable {
+    vim = {
+      startPlugins = [
+        "plenary-nvim"
+        "nvim-treesitter"
```

Comment:
There isn't a toggle for plenary, but you may instead explicitly enable the treesitter module when this plugin is enabled. That should be preferred over putting dependencies in startPlugins in ways that cannot be overriden.